lua/crates/semver.lua

```diff
@@ -200,22 +200,112 @@ function M.parse_requirements(str)
     return requirements
 end
 
--- TODO: port https://github.com/dtolnay/semver/blob/master/src%2Fimpls.rs#L51-L107
 ---@param version? string
 ---@param req? string
 ---@return integer
 function M.compare_pre(version, req)
-    if version and req then
-        if version < req then
+    -- Handle identical strings (optimization)
+    if version == req then
+        return 0
+    end
+
+    -- Handle empty prerelease cases
+    local version_empty = not version or version == ""
+    local req_empty = not req or req == ""
+
+    if version_empty and req_empty then
+        return 0
+    elseif version_empty then
+        -- A real release compares greater than prerelease
+        return 1
+    elseif req_empty then
+        -- Prerelease compares less than the real release
+        return -1
+    end
+
+    -- Both are non-empty prerelease strings, compare them
+    ---@param str string
+    ---@return string[]
+    local function split_identifiers(str)
+        local parts = {}
+        for part in str:gmatch("[^%.]+") do
+            table.insert(parts, part)
+        end
+        return parts
+    end
+
+    ---@param str string
+    ---@return boolean
+    local function is_numeric(str)
+        return str:match("^%d+$") ~= nil
+    end
+
+    ---@param a string
+    ---@param b string
+    ---@return integer
+    local function compare_numeric_strings(a, b)
+        -- First compare by length (shorter number is smaller)
+        local len_cmp = #a - #b
+        if len_cmp ~= 0 then
+            return len_cmp
+        end
+        -- If same length, compare lexically
+        if a < b then
             return -1
-        elseif version == req then
+        elseif a > b then
+            return 1
+        else
             return 0
-        elseif version > req then
+        end
+    end
+
+    local version_parts = version and split_identifiers(version) or {}
```

Comment:
I think this can be just `vim.split(version, ".")`, and then the `split_identifiers` function can be removed.